src/WebDAVAdapter.php

```diff
@@ -400,8 +401,18 @@ protected function normalizeObject(array $object, $path)
         return $result;
     }
 
+    /**
+     * @param array $object
+     * @return bool
+     */
     protected function isDirectory(array $object)
     {
-        return isset($object['{DAV:}iscollection']) && $object['{DAV:}iscollection'] === '1';
```

Comment:
wouldn't it make more sense to default to the current implementation instead of returning false in this case?

Reply:
A good question :thinking: `{DAV:}iscollection` is part of [Web Distributed Authoring and Versioning (WebDAV) Protocol: Microsoft Extensions](https://docs.microsoft.com/en-us/openspecs/sharepoint_protocols/ms-wdvme/6d81c2e3-9893-4c1d-9e57-4d2345c625f2). Microsoft Extensions sounds like WebDav + additional features from Microsoft.

> Specifies the Web Distributed Authoring and Versioning (WebDAV) Protocol: Microsoft Extensions, which extends the WebDAV Protocol specified in [RFC2518] to extend the standard Hypertext Transfer Protocol (HTTP) mechanisms specified in [RFC2616] for providing file access and content management over the Internet.

As described on [this page](https://docs.microsoft.com/en-us/openspecs/sharepoint_protocols/ms-wdvme/8cafdf55-ee5c-443e-bdb7-2cb2ab1fb2c3) MS-WDVME extends [RFC2518](https://tools.ietf.org/html/rfc2518) and `{DAV:}resourcetype` is not part of it but RFC2518 has been replaced by [RFC4918](https://tools.ietf.org/html/rfc4918). I would assume that `{DAV:}resourcetype` work with sharepoint but can't test. 

> wouldn't it make more sense to default to the current implementation instead of returning false in this case?

I would say no. We can still add a `MicrosoftWebDAVAdapter` if people are asking for it. What do you think?

Reply:
@frankdejonge  Please let me know if there's anything we need to change.

Reply:
With Nextcloud and Flysystem and flysystem-webdav, the implementation do not work anymore.
It may start with Microsoft, but is now used in many implementations.

Reply:
Could you elaborate? Are you voting against this change because other implementations started to adopt this?

Reply:
At least yes, if this is possible. I'm with frankdejonge on this.
Is there no possibility to get it backward compatible to 1.0.5 (my last working version)?

Reply:
If you apply this patch it's not possible to mount a webdav storage provided by nextcloud? I doubt that because my motivation for this change was this ticket at nextcloud: https://github.com/nextcloud/server/issues/16267

Nextcloud does not support `iscollection`. I see your point but webdav is rfc4918. Take a look at https://github.com/thephpleague/flysystem-webdav/pull/56. I've added some tests against a real webdav server. Tests are failing because the directory detection is not compliant with rfc4918.

I'm fine if you don't want this change. You have to care about the users and breaking existing installations should be avoided.

Reply:
@JanFischer2 https://github.com/thephpleague/flysystem-webdav/compare/1.0.5...1.0.6 if 1.0.5 is the latest working version than it's exactly the `iscollection` change (which has been released with 1.0.6) that breaks your setup. So you should actually vote for this ;)

Reply:
Ah ok. Now i understand. No vote so far.
Ok I will have a deeper look in my nextcloud<->flysystem communication.

Reply:
Yes you are right an I did not read well enough.
Or better, I still thought in the wrong way.
This change works fine.
Thank you and have a nice weekend :)

Reply:
So now the second part make more sence to me.
Do somebody have a share point webdav?
Does share point provide both attributes or just iscollection?

Cant you still request iscollection and resourcetype?
If its iscollection not set, you can check for resourcetype.

Reply:
> Cant you still request iscollection and resourcetype?

Yes but `iscollection` is not a part of rfc4918. The proper way to check for a directory is `resourcetype`. I added an example how to keep support for the Microsoft stuff. If someone needs it he can easly adopt it. There is no way to setup a share point instance for testing (so no automated tests). Right now there is only one attribute of this [MS-WDVME](https://docs.microsoft.com/en-us/openspecs/sharepoint_protocols/ms-wdvme/8cafdf55-ee5c-443e-bdb7-2cb2ab1fb2c3) supported but there is more. Look we needed jquery for years because browser vendors were not able (or willing) to keep their products inter operable.

Reply:
Yeah. I fully agree with you. 
Thats the old web implementation point; especially IE.
The other site is, that it is as it is.
I had this disusion on my oown and with other, many times ;)
May be do it the correct way, but if has no resourcetype, check iscollection.
Its not the fault of the user, what microsoft has done.
But only as idea from me.

Reply:
Fine by me.